api/src/main/java/com/grappus/api/cache/AuthStore.kt

```diff
@@ -41,5 +41,9 @@ class AuthStore @Inject constructor(
         }
 
     val isLoggedIn: Boolean
-        get() = !authToken.isNullOrEmpty() && user != null
+        get() = user.isNotNull()
+
+    private fun User?.isNotNull(): Boolean {
+        return !authToken.isNullOrEmpty() && user != null && this?.userId != 0L
+    }
```

Comment:
Added this here because api module cannot access any other module. May be we can move this below the data class itself. I was not sure.